### PR TITLE
Refactor: Update docs site to use new `<bolt-code-snippet>` component

### PR DESCRIPTION
**Pull request by sghoweri**
Wires up the new `<bolt-code-snippet>` component to our PL docs + removes the original (temporary) highlight.js syntax highlighting solution originally added.

### CODE REVIEW

File: apps/pattern-lab/src/_includes/utils/docs.twig [hunk 1]
```diff
@@ -46,7 +46,11 @@
           id: "usage"
         }
       } %}
-      <pre class="u-bolt-margin-bottom-large"><code>{{ code }}</code></pre>
+      {% spaceless %}
+      <bolt-code-snippet display="block" lang="twig">
+          <pre><code is="shadow-root">{{ code }}</code></pre>
+      </bolt-code-snippet>
```
[the remark this answers is not in the document]
Comment (sghoweri) [reply apps/pattern-lab/src/_includes/utils/docs.twig]: @remydenton there's actually two reasons why I have this in here:

1. Make sure we have a working solution that doesn't require Twig yet still visually works all the same as the one that does (which I get that in this particular file isn't an issue but ties in with my 2nd point). For example, testing this out with and without JavaScript enabled:
Without JS (left) vs With JS (right)
image

2. Markdown files (which most of the docs are based off of) can't support embedded Twig so we need a solution that compiles ok AND still contains semantic <code> and <pre> tags so this thing renders as expected on NPM and Github sites when the component's JS isn't firing.

[the remark this answers is not in the document]
Comment (sghoweri) [reply apps/pattern-lab/src/_includes/utils/docs.twig]: @remy

3. Also keep in mind that we CAN use the `<bolt-code-snippet>` the way you were mentioning. 

The main reason why this isn’t always ideal in every environment (besides when getting used in something like Markdown that isn’t pre-compiled) is that you get a flash of unstylized text before this component renders. 

I’m planning ahead for us to get the docs site to eventually be fully JS powered (and server-side pre-rendered) however that’ll take a couple steps to get us there. 

This is one of those steps.

[the remark this answers is not in the document]
Comment (mikemai2awesome) [reply apps/pattern-lab/src/_includes/utils/docs.twig]: I have to agree with @remydenton, let's use twig until we can use the web component without any tricks. We don't need to introduce problems to the doc site.

**Comment by sghoweri** [comment]
Merging this in since all we’ve got left here is getting the docs site to not load up duplicate syntax highlighting CSS and JS. +1 for web performance!

**Pull request merged**